### PR TITLE
Use blueprintjs InputGroup and TextArea

### PR DESCRIPTION
**Pull request by wadjih-bencheikh18**
closes : https://github.com/zakodium-oss/react-science/issues/547

### CODE REVIEW

**Comment by stropitek** [comment]
As a part of this PR you can update the form story (https://547-extend-the-input-compone.react-science.pages.dev/stories/?path=/story/forms-form--control) with all possible form elements (input, textarea, checkbox, select, radio etc...)

Use the `FormGroup` component to add labels and helper text.

Throw in a native HTML select (https://blueprintjs.com/docs/#core/components/html-select)

**Comment by wadjih-bencheikh18** [comment]
@stropitek I couldn't understand build is failing here

**Comment by stropitek** [comment]
@wadjih-bencheikh18 this should be fixed if you rebase / merge main into this branch

File: stories/components/form.stories.tsx [hunk 1]
```diff
@@ -28,16 +20,21 @@ export function Control() {
...
-      </Field>
-      <Field label="Status" name="status">
+      <FormGroup
+        label="Name"
+        labelFor="name"
+        labelInfo={<span style={{ color: 'red' }}>*</span>}
+        intent="danger"
+      >
+        <InputGroup type="text" maxLength={10} />
+      </FormGroup>
```
Comment (stropitek) [inline stories/components/form.stories.tsx:30]: Forms inputs have missing `id` attributes to link with `labelFor` and enable focus of input when clicking the label.

Please check all stories for this.

File: stories/components/form.stories.tsx [hunk 2]
```diff
@@ -28,16 +20,21 @@ export function Control() {
   const [radio, setRadio] = useState<RadioOption | undefined>(undefined);
   return (
     <ExampleGroup style={{ display: 'inline-block' }}>
```
Comment (stropitek) [inline stories/components/form.stories.tsx:22]: there are conflicting css rules on `ExampleGroup` and in the `style` prop.

File: stories/components/input.stories.tsx [hunk 3]
```diff
@@ -12,54 +18,64 @@ const ExampleContainerAddonGroup = styled.div`
...
+  onChange: { action: 'onChange' },
+  placeholder: { control: 'text' },
+  leftIcon: {
+    control: { type: 'inline-radio' },
+    options: [undefined, 'person', 'shield', 'add', 'lightning'],
+  },
+  rightIcon: {
+    control: { type: 'inline-radio' },
+    options: [undefined, 'person', 'shield', 'add', 'lightning'],
+  },
```
Comment (stropitek) [inline stories/components/input.stories.tsx:70]: This doesn't work and I believe it can be removed

**Review by stropitek — COMMENTED** [review]
The stories with the indeterminate checkboxes don't work as before. The checkbox should be indeterminate when only some options are selected.
CleanShot 2023-11-27 at 11 24 17
CleanShot 2023-11-27 at 11 23 45

---

Input stories: only keep the control and form group control story, the other stories can be deleted.

Form group control has missing controls, for example disabled, intent etc... Please review to add as many controls as possible.

---

I spotted other problems in the stories but I'll stop the review here for now. Please carefully look at the stories, especially the ones with control to check if they all work and don't have missing controls.

**Review by stropitek — APPROVED** [review]
I'm merging this and creating separate issues for the remaining issues.

**Pull request merged**